### ISSUE TITLE
struct2depth: Pre-computed mask, Online refinement and Pose estimation

### ISSUE THREAD

**Issue by player1321** [tensorflow/models#7410]
### System information
- **What is the top-level directory of the model you are using**: tensorflow/models/struct2depth/
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Only paths are changed (details below)
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: miniconda2
- **TensorFlow version (use command below)**: 1.13.1
- **Bazel version (if compiling from source)**: no
- **CUDA/cuDNN version**:  release 9.0, V9.0.176
- **GPU model and memory**: 1080Ti 12GB
- **Exact command to reproduce**: below is the command I run optimize.py

> prediction_dir=/path/to/inference/result
> model_ckpt=/path/to/ckpt
> handle_motion="true"
> size_constraint_weight="0.0005"
> 
> data_dir=/path/to/kitti/data
> triplet_list_file=$data_dir/test_files_eigen_triplets.txt
> triplet_list_file_remains=$data_dir/test_files_eigen_triplets_remains.txt
> ft_name=kitti
> 
> python optimize.py \
>   --output_dir $prediction_dir \
>   --data_dir $data_dir \
>   --triplet_list_file $triplet_list_file \
>   --triplet_list_file_remains $triplet_list_file_remains \
>   --ft_name $ft_name \
>   --model_ckpt $model_ckpt \
>   --file_extension png \
>   --size_constraint_weight $size_constraint_weight

### Describe the problem
1. Pre-computed mask: I use pretrained Mask-RCNN to generate the masks. But the model was not improved when training with these masks. Obvious gaps can be seen during training. In your paper you said the mask helps to handle moving objects. So how do you select masks for moving objects? Since Mask-RCNN only provide masks for specific categories of objects, no matter they are moving or not.
![image](https://user-images.githubusercontent.com/38416786/62671089-0f221380-b9c8-11e9-97bd-aa367ae0ecf8.png)

2. Online refinement: I ran optimize.py but both the R mode and M+R mode got no improvements. I generated `triplet_list_file.txt` according to your code (part of it can be seen below). It can be very helpful if you can provide more details about how to prepare the input files for optimization.
> 2011_09_26/2011_09_26_drive_0002_sync/image_02/data 0000000069
> 2011_09_26/2011_09_26_drive_0002_sync/image_02/data 0000000054
> 2011_09_26/2011_09_26_drive_0002_sync/image_02/data 0000000042
> 2011_09_26/2011_09_26_drive_0002_sync/image_02/data 0000000057
> 2011_09_26/2011_09_26_drive_0002_sync/image_02/data 0000000030

3. Pose estimation: I tried to evaluate the output of the pose net, but it seemed that there should be some operations to transform the output of pose net to trajectory like that provided by KITTI Odometry ground truth. Can you provide some details about the meaning of the output of pose net? Or some script to help transform the raw output to predicted trajectory?

@VincentCa
@aneliaangelova
All helps or replys are highly appreciated!
Thank you very much!

Best,
